From f76febbc8a24d729078933f546ef9cfd3727bf42 Mon Sep 17 00:00:00 2001
From: John Jacques <john.jacques@intel.com>
Date: Fri, 15 May 2015 19:10:26 -0500
Subject: [PATCH 032/213] arch/arm64: Correct physical addresses for Simulation
 dts

Prior to this commit, the simulation environment used different
physical addresses for the GIC than the hardware (X9). After this
commit, the addresses match.

Signed-off-by: John Jacques <john.jacques@intel.com>
---
 arch/arm64/boot/dts/intel/Makefile          |   2 +-
 arch/arm64/boot/dts/intel/axm5604-sim.dts   |   2 +-
 arch/arm64/boot/dts/intel/axm5608-cpus.dtsi |  30 +++++-
 arch/arm64/boot/dts/intel/axm5608-sim.dts   | 152 ++++++++++++++++++++++++++++
 arch/arm64/boot/dts/intel/axm5616-cpus.dtsi | 129 +++++++++++++++++++++++
 arch/arm64/boot/dts/intel/axm5616-sim.dts   | 152 ++++++++++++++++++++++++++++
 arch/arm64/boot/dts/intel/axm56xx-sim.dtsi  |  12 ++-
 arch/arm64/boot/dts/intel/axm56xx.dtsi      |   4 +-
 8 files changed, 473 insertions(+), 10 deletions(-)
 create mode 100644 arch/arm64/boot/dts/intel/axm5608-sim.dts
 create mode 100644 arch/arm64/boot/dts/intel/axm5616-cpus.dtsi
 create mode 100644 arch/arm64/boot/dts/intel/axm5616-sim.dts

diff --git a/arch/arm64/boot/dts/intel/Makefile b/arch/arm64/boot/dts/intel/Makefile
index 1dd6d29..5c6e98c 100644
--- a/arch/arm64/boot/dts/intel/Makefile
+++ b/arch/arm64/boot/dts/intel/Makefile
@@ -1,4 +1,4 @@
-dtb-$(CONFIG_ARCH_AXXIA) += axm5604-sim.dtb axm5601-emu.dtb axm5602-emu.dtb axm5606-emu.dtb axc6704-sim.dtb
+dtb-$(CONFIG_ARCH_AXXIA) += axm5604-sim.dtb axm5608-sim.dtb axm5616-sim.dtb axm5601-emu.dtb axm5602-emu.dtb axm5606-emu.dtb axc6704-sim.dtb
 
 always		:= $(dtb-y)
 subdir-y	:= $(dts-dirs)
diff --git a/arch/arm64/boot/dts/intel/axm5604-sim.dts b/arch/arm64/boot/dts/intel/axm5604-sim.dts
index 38cdbfc..287957b 100644
--- a/arch/arm64/boot/dts/intel/axm5604-sim.dts
+++ b/arch/arm64/boot/dts/intel/axm5604-sim.dts
@@ -11,7 +11,7 @@
 
 /dts-v1/;
 
-#include "axm56xx-sim.dtsi"
+#include "axm56xx.dtsi"
 #include "axm5604-cpus.dtsi"
 
 / {
diff --git a/arch/arm64/boot/dts/intel/axm5608-cpus.dtsi b/arch/arm64/boot/dts/intel/axm5608-cpus.dtsi
index db31656..96b3cea 100644
--- a/arch/arm64/boot/dts/intel/axm5608-cpus.dtsi
+++ b/arch/arm64/boot/dts/intel/axm5608-cpus.dtsi
@@ -28,7 +28,7 @@
 			enable-method = "psci";
 		};
 
-	      cpu@2 {
+		cpu@2 {
 			device_type = "cpu";
 			compatible = "arm,armv8";
 			reg = <0x02>;
@@ -41,5 +41,33 @@
 			reg = <0x03>;
 			enable-method = "psci";
 		};
+
+		cpu@4 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x100>;
+			enable-method = "psci";
+		};
+
+		cpu@5 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x101>;
+			enable-method = "psci";
+		};
+
+		cpu@6 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x102>;
+			enable-method = "psci";
+		};
+
+		cpu@7 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x103>;
+			enable-method = "psci";
+		};
 	};
 };
diff --git a/arch/arm64/boot/dts/intel/axm5608-sim.dts b/arch/arm64/boot/dts/intel/axm5608-sim.dts
new file mode 100644
index 0000000..e9302f4
--- /dev/null
+++ b/arch/arm64/boot/dts/intel/axm5608-sim.dts
@@ -0,0 +1,152 @@
+/*
+ * arch/arm64/boot/dts/intel/axm5604-sim.dts
+ *
+ * Copyright (C) 2013 LSI
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+/dts-v1/;
+
+#include "axm56xx.dtsi"
+#include "axm5608-cpus.dtsi"
+
+/ {
+	model = "AXM56xx Simulator";
+	compatible = "lsi,axm5616";
+
+	memory {
+		device_type = "memory";
+		reg = <0 0x00000000 0x40000000>;
+	};
+
+	mmc_3v3: fixedregulator@0 {
+		compatible = "regulator-fixed";
+		regulator-name = "3V3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+	};
+
+	soc {
+		virtio_block@8021000000 {
+			compatible = "virtio,mmio";
+			reg = <0x80 0x21000000 0x10000>;
+			interrupts = <GIC_SPI 279 IRQ_TYPE_LEVEL_HIGH>;
+		};
+
+		vmfs@8021010000 {
+			compatible = "arm,messagebox";
+			reg = <0x80 0x21010000 0x1000>;
+			interrupts = <GIC_SPI 278 IRQ_TYPE_LEVEL_HIGH>;
+		};
+
+		ethernet@8020000000 {
+			compatible = "smsc,lan91c111";
+			reg = <0x80 0x20000000 0x1000000>;
+			interrupts = <GIC_SPI 280 IRQ_TYPE_LEVEL_HIGH>;
+			phy-mode = "mii";
+			reg-io-width = <4>;
+			smsc,irq-active-high;
+			smsc,irq-push-pull;
+		};
+
+		amba {
+			mmci@8021030000 {
+				compatible = "arm,pl18x", "arm,primecell";
+				reg = <0x80 0x21030000 0x10000>;
+				interrupts = <GIC_SPI 281 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 282 IRQ_TYPE_LEVEL_HIGH>;
+				vmmc-supply = <&mmc_3v3>;
+				vqmmc-supply = <&mmc_3v3>;
+				clocks = <&clks 0>, <&clks 1>;
+				clock-names = "apb_pclk", "mmc_clk";
+			};
+		};
+	};
+};
+
+&serial0 {
+	status = "okay";
+};
+
+&serial1 {
+	status = "okay";
+};
+
+&serial2 {
+	status = "okay";
+};
+
+&serial3 {
+	status = "okay";
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "s25fl016k";
+		reg = <0x0>;
+		spi-max-frequency = <25000000>;
+		pl022,com-mode = <1>;
+
+		partition@0 {
+			label = "bl2a";
+			reg = <0x0 0x40000>;
+		};
+		partition@40000 {
+			label = "bl2b";
+			reg = <0x40000 0x40000>;
+		};
+	};
+};
+
+&i2c0 {
+	status = "okay";
+};
+
+&i2c1 {
+	status = "okay";
+};
+
+&i2c2 {
+	status = "okay";
+};
+
+&i2c3 {
+	status = "okay";
+};
+
+&gpdma0 {
+	status = "okay";
+};
+
+&gpdma1 {
+	status = "okay";
+};
+&pci0 {
+
+	status = "okay";
+};
+
+&pci1 {
+	status = "okay";
+};
+
+&pci2 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/intel/axm5616-cpus.dtsi b/arch/arm64/boot/dts/intel/axm5616-cpus.dtsi
new file mode 100644
index 0000000..da1dd40
--- /dev/null
+++ b/arch/arm64/boot/dts/intel/axm5616-cpus.dtsi
@@ -0,0 +1,129 @@
+/*
+ * arch/arm64/boot/dts/intel/axm5616-cpus.dtsi
+ *
+ * Copyright (C) 2014 Avago Technologies
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x00>;
+			enable-method = "psci";
+		};
+
+		cpu@1 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x01>;
+			enable-method = "psci";
+		};
+
+		cpu@2 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x02>;
+			enable-method = "psci";
+		};
+
+		cpu@3 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x03>;
+			enable-method = "psci";
+		};
+
+		cpu@4 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x100>;
+			enable-method = "psci";
+		};
+
+		cpu@5 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x101>;
+			enable-method = "psci";
+		};
+
+		cpu@6 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x102>;
+			enable-method = "psci";
+		};
+
+		cpu@7 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x103>;
+			enable-method = "psci";
+		};
+
+		cpu@8 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x200>;
+			enable-method = "psci";
+		};
+
+		cpu@9 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x201>;
+			enable-method = "psci";
+		};
+
+		cpu@10 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x202>;
+			enable-method = "psci";
+		};
+
+		cpu@11 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x203>;
+			enable-method = "psci";
+		};
+
+		cpu@12 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x300>;
+			enable-method = "psci";
+		};
+
+		cpu@13 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x301>;
+			enable-method = "psci";
+		};
+
+		cpu@14 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x302>;
+			enable-method = "psci";
+		};
+
+		cpu@15 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			reg = <0x303>;
+			enable-method = "psci";
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/intel/axm5616-sim.dts b/arch/arm64/boot/dts/intel/axm5616-sim.dts
new file mode 100644
index 0000000..46e170f
--- /dev/null
+++ b/arch/arm64/boot/dts/intel/axm5616-sim.dts
@@ -0,0 +1,152 @@
+/*
+ * arch/arm64/boot/dts/intel/axm5616-sim.dts
+ *
+ * Copyright (C) 2013 LSI
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+/dts-v1/;
+
+#include "axm56xx.dtsi"
+#include "axm5616-cpus.dtsi"
+
+/ {
+	model = "AXM56xx Simulator";
+	compatible = "lsi,axm5616";
+
+	memory {
+		device_type = "memory";
+		reg = <0 0x00000000 0x40000000>;
+	};
+
+	mmc_3v3: fixedregulator@0 {
+		compatible = "regulator-fixed";
+		regulator-name = "3V3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+	};
+
+	soc {
+		virtio_block@8021000000 {
+			compatible = "virtio,mmio";
+			reg = <0x80 0x21000000 0x10000>;
+			interrupts = <GIC_SPI 279 IRQ_TYPE_LEVEL_HIGH>;
+		};
+
+		vmfs@8021010000 {
+			compatible = "arm,messagebox";
+			reg = <0x80 0x21010000 0x1000>;
+			interrupts = <GIC_SPI 278 IRQ_TYPE_LEVEL_HIGH>;
+		};
+
+		ethernet@8020000000 {
+			compatible = "smsc,lan91c111";
+			reg = <0x80 0x20000000 0x1000000>;
+			interrupts = <GIC_SPI 280 IRQ_TYPE_LEVEL_HIGH>;
+			phy-mode = "mii";
+			reg-io-width = <4>;
+			smsc,irq-active-high;
+			smsc,irq-push-pull;
+		};
+
+		amba {
+			mmci@8021030000 {
+				compatible = "arm,pl18x", "arm,primecell";
+				reg = <0x80 0x21030000 0x10000>;
+				interrupts = <GIC_SPI 281 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 282 IRQ_TYPE_LEVEL_HIGH>;
+				vmmc-supply = <&mmc_3v3>;
+				vqmmc-supply = <&mmc_3v3>;
+				clocks = <&clks 0>, <&clks 1>;
+				clock-names = "apb_pclk", "mmc_clk";
+			};
+		};
+	};
+};
+
+&serial0 {
+	status = "okay";
+};
+
+&serial1 {
+	status = "okay";
+};
+
+&serial2 {
+	status = "okay";
+};
+
+&serial3 {
+	status = "okay";
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "s25fl016k";
+		reg = <0x0>;
+		spi-max-frequency = <25000000>;
+		pl022,com-mode = <1>;
+
+		partition@0 {
+			label = "bl2a";
+			reg = <0x0 0x40000>;
+		};
+		partition@40000 {
+			label = "bl2b";
+			reg = <0x40000 0x40000>;
+		};
+	};
+};
+
+&i2c0 {
+	status = "okay";
+};
+
+&i2c1 {
+	status = "okay";
+};
+
+&i2c2 {
+	status = "okay";
+};
+
+&i2c3 {
+	status = "okay";
+};
+
+&gpdma0 {
+	status = "okay";
+};
+
+&gpdma1 {
+	status = "okay";
+};
+&pci0 {
+
+	status = "okay";
+};
+
+&pci1 {
+	status = "okay";
+};
+
+&pci2 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/intel/axm56xx-sim.dtsi b/arch/arm64/boot/dts/intel/axm56xx-sim.dtsi
index e8a4495..adaa0a5 100644
--- a/arch/arm64/boot/dts/intel/axm56xx-sim.dtsi
+++ b/arch/arm64/boot/dts/intel/axm56xx-sim.dtsi
@@ -56,8 +56,8 @@
 		#interrupt-cells = <3>;
 		#address-cells = <0>;
 		interrupt-controller;
-		reg = <0x80 0x10010000 0x10000>,
-		      <0x80 0x10100000 0x200000>;
+		reg = <0x80 0x10000000 0x10000>,
+		      <0x80 0x10200000 0x200000>;
 		interrupts = <GIC_PPI 9 IRQ_TYPE_LEVEL_HIGH>;
 	};
 
@@ -101,9 +101,11 @@
 		};
 
 		nemac: ethernet@8080240000 {
-			compatible = "lsi,nemac";
-			reg = <0x80 0x80240000 0x1000>;
-			interrupts = <GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>;
+			compatible = "intel,nemac";
+			reg = <0x80 0x80240000 0x10000>;
+			interrupts = <GIC_SPI 0 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 1 IRQ_TYPE_LEVEL_HIGH>;
+			dma-coherent;
 			status = "disabled";
 		};
 
diff --git a/arch/arm64/boot/dts/intel/axm56xx.dtsi b/arch/arm64/boot/dts/intel/axm56xx.dtsi
index 6dc498a..63aa8fa 100644
--- a/arch/arm64/boot/dts/intel/axm56xx.dtsi
+++ b/arch/arm64/boot/dts/intel/axm56xx.dtsi
@@ -83,8 +83,8 @@
 		ranges;
 
 		syscon: syscon@8002000000 {
-			compatible = "lsi,axxia-syscon", "syscon";
-			reg = <0x80 0x02000000 0x40000>;
+			compatible = "intel,axxia-syscon", "syscon";
+			reg = <0x80 0x02c00000 0x40000>;
 		};
 
 		reset: reset@2010031000 {
-- 
1.9.1

